test/CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ find_package(Catch2 REQUIRED)
 include(CTest)
 include(Catch)
 
-add_library(catch_main STATIC catch_main.cpp)
+add_library(catch_main OBJECT catch_main.cpp)
 target_link_libraries(catch_main PUBLIC Catch2::Catch2)
 target_link_libraries(catch_main PRIVATE project_options)
 
```

Comment:
We should not use XML as the default reporter. This is very hard to read from CLI. XML should be used only for coverage. 

In general, I prefer the default settings of catch2 when coverage is off.
```cmake
catch_discover_tests(tests)
```

Reply:
Are we actually setting a default reporter anywhere? 

This change I think is for the better regardless, because it is more appropriate for the use of catch_main that we need. We don't really need a library target, we just need to not build it more than one time.

Reply:
This can be addressed in another PR. I also like to move the scripts for analyzing coverage using gcovr and opencppcoverage to the CMake file

Reply:
> Are we actually setting a default reporter anywhere?

Yeah. A few lines below this line. I could not leave my comment there due to GitHub limitations. This isn't really related to this PR.